### ISSUE TITLE
Port _initgroups.c to 3.1

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4248]
|<img alt="loewis's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| loewis reported|
|-|-|
|Trac ID|trac#4248|
|Type|enhancement|
|Created|2010-01-30 18:25:44Z|
|Branch|https://github.com/twisted/twisted/tree/initgroups-3k-4248|

The port supports both 2.x and 3.x.

Attachments:

* [initgroups.diff](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/3fa/3faf9252ee29ed55fee7ca30e9926d67a12e4384/8287511500c3606a8ae00e248c476ecf4d6ed21f.diff) (641 bytes) - added by loewis on 2010-01-30 18:25:58Z - 
* [4248.feature](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/3fa/3faf9252ee29ed55fee7ca30e9926d67a12e4384/134d699fe326fded2f88f5770dc4a03719fc41ee.feature) (434 bytes) - added by loewis on 2010-01-31 08:13:07Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__4248 4248
type__enhancement enhancement
reporter__loewis loewis
priority__normal normal
milestone__Python_3_x Python-3.x
branch__branches_initgroups_3k_4248 branches/initgroups-3k-4248
branch_author__therve therve
status__closed closed
resolution__wontfix wontfix
component__core core
keywords__py3k py3k
time__1264875944000000 1264875944000000
changetime__1291711357000000 1291711357000000
version__None None
owner__ 
cc__thijs
```
</details>

**Comment by twisted-trac** [comment]
|<img alt="loewis's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|loewis removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_3"></a>|@thijstriemstra commented|
|-|-|

Blocked by [#4244](https://github.com/twisted/twisted/issues/4244)

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)|@therve set owner to @therve|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_5"></a>|@therve commented|
|-|-|

(In [28252]) Branching to 'initgroups-3k-4248'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_7"></a>|@exarkun set status to `closed`|
|-|-|

The initgroups wrapper has been pushed into the standard library.  I don't know if it made it into 3.1, but it will be in 3.2.  At this point, it seems unlikely that anyone will be using 3.1 by the time Twisted gets to 3.x, so maintaining a 3.x port of this C code doesn't make much sense.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
